server/accounts.go

```diff
@@ -2651,7 +2684,329 @@ func (ur *URLAccResolver) Fetch(name string) (string, error) {
+		if len(tk) != accUpdateTokens {
+			return
+		}
+		pubKey := tk[accUpdateAccIndex]
+		if claim, err := jwt.DecodeAccountClaims(string(msg)); err != nil {
+			respondToUpdate(s, resp, pubKey, "jwt update resulted in error", err)
+		} else if claim.Subject != pubKey {
+			err := errors.New("subject does not match jwt content")
+			respondToUpdate(s, resp, pubKey, "jwt update resulted in error", err)
+		} else if err := dr.save(pubKey, string(msg)); err != nil {
+			respondToUpdate(s, resp, pubKey, "jwt update resulted in error", err)
+		} else {
+			respondToUpdate(s, resp, pubKey, "jwt updated", nil)
+		}
+	}); err != nil {
+		s.Errorf("Error setting up update handling: %v", err)
+	}
+	// respond to lookups with our version
+	if subLookupReq, err = s.sysSubscribe(accountLookupRequest, func(_ *subscription, _ *client, subj, reply string, msg []byte) {
+		if reply == "" {
+			return
+		}
+		tk := strings.Split(subj, tsep)
+		if len(tk) != accUpdateTokens {
+			return
+		}
+		if theJWT, err := dr.DirJWTStore.LoadAcc(tk[accUpdateAccIndex]); err != nil {
+			s.Errorf("Merging resulted in error: %v", err)
+		} else {
+			s.sendInternalMsgLocked(reply, "", nil, []byte(theJWT))
+		}
+	}); err != nil {
+		s.Errorf("Error setting up pack response handling: %v", err)
+	}
+	// respond to pack requests with one or more pack messages
+	// an empty message signifies the end of the response responder
+	if subPackReq, err = s.sysSubscribeQ(accountPackRequest, "responder",
+		func(_ *subscription, _ *client, _, reply string, theirHash []byte) {
+			if reply == "" {
+				return
+			}
+			ourHash := dr.DirJWTStore.Hash()
+			if bytes.Equal(theirHash, ourHash[:]) {
+				s.sendInternalMsgLocked(reply, "", nil, []byte{})
+				s.Debugf("pack request matches hash %x", ourHash[:])
+			} else if msg, err := dr.DirJWTStore.Pack(-1); err != nil {
+				// let them timeout
+				s.Errorf("pack request error: %v", err)
+			} else {
+				s.Debugf("pack request hash %x - respond with hash %x / %d bytes", theirHash, ourHash, len(msg))
+				s.sendInternalMsgLocked(reply, "", nil, []byte(msg))
+				s.sendInternalMsgLocked(reply, "", nil, []byte{})
+			}
+		}); err != nil {
+		s.Errorf("Error setting up pack request handling: %v", err)
+	}
+	// embed pack responses into store
+	packRespIb := s.newRespInbox()
+	if subPackResp, err = s.sysSubscribe(packRespIb, func(_ *subscription, _ *client, _, _ string, msg []byte) {
+		hash := dr.DirJWTStore.Hash()
+		if len(msg) == 0 { // end of response stream
+			s.Debugf("Merging Finished and resulting in: %x", dr.DirJWTStore.Hash())
+			return
+		} else if err := dr.DirJWTStore.Merge(string(msg)); err != nil {
+			s.Errorf("Merging resulted in error: %v", err)
+		} else {
+			s.Debugf("Merging succeeded and changed %x to %x", hash, dr.DirJWTStore.Hash())
+		}
+	}); err != nil {
+		s.Errorf("Error setting up pack response handling: %v", err)
+	}
+	// periodically send out pack message
+	quit := make(chan struct{})
+	go func() {
```

Comment:
Should this be plumbed into the go routine handlers such that the server can clean these up?

Reply:
quit is retained by the returned close function.
When the server calls that it, the go routine will be cleaned up.

Reply:
ok why the different way to cleanup when there is a common way to manage server based Go routines? Just curious.

Reply:
Start is called in setSystemAccount. 
There we start the go routine without the use of s.startGoRoutine.
Eventing is also shut down separately from other code. (prior to it)
So I operated under the same assumptions that I need to be able to stop what was started. 
(vs. shut everything down after eventing has already been shutdown)

Returning a closer seemed like a good way to do just that. 
(IMHO, for infrequent operations this is a nice pattern)

Reply:
Might be good to coordinate with @kozlovic when he is back on his generic handling of Go routines in the server.

Reply:
Yes, will bring this up with him